module/Database/src/Model/Member.php

```diff
@@ -367,9 +357,47 @@ public function setLidnr(int $lidnr): void
     /**
      * Set the member's email address.
      */
-    public function setEmail(?string $email): void
+    public function setEmail(?string $newEmail): void
     {
-        $this->email = $email;
+        // If the new email address matches the current, we don't have to do anything
+        if ($this->email === $newEmail) {
+            return;
+        }
+
+        $oldEmail = $this->email;
+        $this->email = $newEmail;
+
+        if (null === $oldEmail) {
+            return;
+        }
+
+        $mailAddressExists = $this->mailingListMemberships->exists(
+            static function ($key, MailingListMember $list) use ($newEmail) {
+                return $newEmail === $list->getEmail();
+            },
+        );
+        if ($mailAddressExists) {
+            throw new RuntimeException(
+                // phpcs:ignore -- user-visible strings should not be split
+                'The e-mail address cannot be updated while there are already (pending) registrations for this member using this email address. Please try again once all list updates have been processed.',
+            );
+        }
```

Comment:
I would have loved to be able to use `wrapInTransaction` here instead, but unfortunately, we do not have access to the `EntityManager` at this point (as this is the model).